src/features/resource-links/ResourceLinkGenerator.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Button, ModalDialog, useToggle } from '@edx/paragon';
+import { currentSequenceSelector } from '../course-player/data/selectors';
+
+const generateResource = (currentSectionId, sequenceId, courseId) => {
+  try {
+    // Get only the section Identifier
+    const sectionString = currentSectionId ? currentSectionId.slice(currentSectionId.lastIndexOf('@') + 1) : '';
+
+    if (!sectionString) {
+      throw new Error('Unable to obtain sectionString');
```

Comment:
did you test this? if I'm not mistaken throwing this would break the whole MFE if there's no general Error Boundary

Reply:
In testing this, the MFE does not break, and the error message appears in the modal as expected